Document benefits from TypeScript annotation (#7423)
stylelint/stylelint: docs/user-guide/configure.md

ES module example:

```js
/** @type {import('stylelint').Config} */
export default {
  rules: {
    "block-no-empty": true
  }
};
```

> [!NOTE]

> > The JSDoc annotation (`@type`) benefits auto-completion or type-check by TypeScript.

Reviewer: ```suggestion
> The `@type` JSDoc annotation enables Typescript to autocomplete and type-check.
```
